transmart-core-db/grails-app/services/org/transmartproject/db/querytool/QueriesResourceService.groovy

```diff
@@ -166,11 +169,60 @@ class QueriesResourceService implements QueriesResource {
         resultInstance
     }
 
+    QueryResult disablingQuery(Long id,
+                               String username) throws InvalidRequestException
+    {
+        QtQueryResultInstance resultInstance = getQueryResultFromId(id)
+
+        if (resultInstance.queryInstance.userId == username) {
+            resultInstance.deleteFlag = "Y"
+        }
+        else{
+            throw new AccessDeniedException("User ${username} has no permission" +
```

Comment:
Usual structure is: 
- first the check and if no permission, throw error
- then the intended action